src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -1760,78 +1755,6 @@ private bool TryGetNullableMethod(SyntaxNode syntax, TypeSymbol nullableType, Sp
             return false;
         }
 
-        private BoundExpression RewriteNullableNullEquality(
```

Comment:
>RewriteNullableNullEquality [](start = 32, length = 27)

I'm curious, why move these methods to SyntheticBoundNodeFactory?

Reply:
The difference between the factory's Binary function and the MakeBinaryOperator in the local rewriter was causing all of my test failures. Binary worked if the BinaryOperatorKind was Equal or NotEqual, but none of the nullables worked. This caused lots of nullable coalescing failures.

So I had to replace MakeBinaryOperator in MakeNullCheck with either Binary or RewriteNullableNullEquality depending on kind.

Reply:
The issue causing problem is in SyntheticBoundNodeFactory.cs, lines 1476- 1483.

Reply:
Consider leaving the methods here, but changing to `static` with an explicit `SyntheticBoundNodeFactory` instance. That should reduce the differences and merge conflicts while this feature is in a branch.

---
In reply to: [299763855](https://github.com/dotnet/roslyn/pull/36832#discussion_r299763855) [](ancestors = 299763855)